src/createWidgetBase.ts

```diff
@@ -212,15 +196,35 @@ const createWidget: WidgetFactory = createStateful
 				return this.properties.id;
 			},
 
-			applyChangedProperties: function(this: Widget<WidgetState, WidgetProperties>, previousProperties: WidgetProperties, currentProperties: WidgetProperties): void {
-				if (Object.keys(currentProperties).length) {
-					currentProperties['id'] = this.id;
-					this.setState(currentProperties);
+			setProperties(this: Widget<WidgetState, WidgetProperties>, properties: WidgetProperties) {
```

Comment:
Is there any reason we didn't make this a setter? i.e. `set properties(this: Widget...)`